tests/test_collate_ab_transforms.py

```diff
@@ -0,0 +1,108 @@
+# ruff: noqa: PLR2004
+
+from pathlib import Path
+
+import pyarrow as pa
+import pytest
+
+from abdiff.core.collate_ab_transforms import (
+    JOINED_DATASET_SCHEMA,
+    READ_BATCH_SIZE,
+    TRANSFORMED_DATASET_SCHEMA,
+    collate_ab_transforms,
+    get_joined_batches_iter,
+    get_transformed_batches_iter,
+    parse_parquet_details_from_transformed_file,
+    yield_records,
+)
+
+
+def test_collate_ab_transforms_success(example_run_directory, example_transformed_files):
+    collated_dataset_path = collate_ab_transforms(
+        run_directory=example_run_directory, transformed_files=example_transformed_files
+    )
+    assert collated_dataset_path == str(Path(example_run_directory) / "collated")
+
+
+def test_yield_records_success(example_transformed_directory):
+    """Validates the structure of the yielded TIMDEX record dictionaries."""
+    records_iter = yield_records(
+        transformed_file=Path(example_transformed_directory)
+        / "a/alma-2024-08-29-daily-transformed-records-to-index.json"
+    )
+    timdex_record_dict = next(records_iter)
+
+    assert list(timdex_record_dict.keys()) == [
+        "timdex_record_id",
+        "source",
+        "record",
+        "version",
+        "transformed_file_name",
+    ]
+    assert isinstance(timdex_record_dict["record"], bytes)
+    assert timdex_record_dict["version"] == "a"
+    assert (
+        timdex_record_dict["transformed_file_name"]
+        == "alma-2024-08-29-daily-transformed-records-to-index"
+    )
+
+
+def test_get_transformed_batches_iter_success(
+    example_run_directory, example_transformed_files
+):
+    transformed_batches_iter = get_transformed_batches_iter(
+        run_directory=example_run_directory, transformed_files=example_transformed_files
+    )
+    transformed_batch = next(transformed_batches_iter)
+
+    assert isinstance(transformed_batch, pa.RecordBatch)
+    assert transformed_batch.num_rows <= READ_BATCH_SIZE
+    assert transformed_batch.schema == TRANSFORMED_DATASET_SCHEMA
+
+
```

Comment:
Arbitrary location of comment: as discussed, probably a good idea to add a test that confirms if **multiple** transform files are provided (e.g. `alma` _and_ `dspace`), that the resulting dataset contains joined rows from both.

Reply:
Updated. Please see latest commit https://github.com/MITLibraries/transmogrifier-ab-diff/pull/22/commits/c8c00da40bb24322d5d28ea4b932895c690d80de.